dask/dataframe/shuffle.py

```diff
@@ -545,15 +565,19 @@ def collect(p, part, meta, barrier_token):
 
 
 def set_partitions_pre(s, divisions):
-    partitions = pd.Series(divisions).searchsorted(s, side="right") - 1
-    partitions[(s >= divisions[-1]).values] = len(divisions) - 2
+    if isinstance(divisions, list) or isinstance(divisions, tuple):
+        divisions = pd.Series(divisions)
+    partitions = divisions.searchsorted(s, side="right") - 1
+    if hasattr(partitions, "index"):
+        partitions = partitions.values
+    partitions[(s >= divisions.iloc[-1]).values] = len(divisions) - 2
     return partitions
 
 
 def shuffle_group_2(df, col):
     if not len(df):
         return {}, df
-    ind = df[col]._values.astype(np.int64)
+    ind = df[col].values.astype(np.int64)
```

Comment:
I'm curious why this was `._values` before.  I wonder if there is some cost in Pandas to calling `.values` rather than `._values`.  @TomAugspurger

Reply:
I'm not sure why it would have originally been `._values`. Here's pandas' table, which should apply to Series too.

```
        It may differ from the public '.values' method.

        index             | values          | _values       | _ndarray_values |
        ----------------- | --------------- | ------------- | --------------- |
        Index             | ndarray         | ndarray       | ndarray         |
        CategoricalIndex  | Categorical     | Categorical   | ndarray[int]    |
        DatetimeIndex     | ndarray[M8ns]   | ndarray[M8ns] | ndarray[M8ns]   |
        DatetimeIndex[tz] | ndarray[M8ns]   | DTI[tz]       | ndarray[M8ns]   |
        PeriodIndex       | ndarray[object] | PeriodArray   | ndarray[int]    |
        IntervalIndex     | IntervalArray   | IntervalArray | ndarray[object] |
```

So `.values` will be slower for things like Period & Interval, it ends up not mattering since we do `.astype(np.int64)` anyway.

Reply:
Thanks @TomAugspurger!  It was indeed my understanding that the behavior of `vaules` should be the same as `_values` in this particular case.  I will take this as reasonable motivation to use `values` in here :)